src/lib/modules/deployment/contract_deployer.js

```diff
@@ -302,7 +302,7 @@ class ContractDeployer {
         next();
       },
       function estimateCorrectGas(next) {
-        if (contract.gas === 'auto') {
+        if (contract.gas === 'auto' || contract.gas === undefined) {
```

Comment:
```suggestion
        if (contract.gas === 'auto' || !contract.gas) {
```
Personal preference. Also covers `null`